homeassistant/components/amcrest/__init__.py

```diff
@@ -160,47 +205,94 @@ def setup(hass, config):
         else:
             authentication = None
 
-        hass.data[DATA_AMCREST][name] = AmcrestDevice(
-            camera, name, authentication, ffmpeg_arguments, stream_source,
+        hass.data[DATA_AMCREST]['devices'][name] = AmcrestDevice(
```

Comment:
Are names always unique?

Reply:
Yes, there's a custom validator to make sure. See [here](https://github.com/pnbruckner/home-assistant/blob/183a07345715e92f5bedd223ba74111b8e042631/homeassistant/components/amcrest/__init__.py#L99-L102) and [here](https://github.com/pnbruckner/home-assistant/blob/183a07345715e92f5bedd223ba74111b8e042631/homeassistant/components/amcrest/__init__.py#L133-L135).